src/lib/DateInput.svelte

```diff
@@ -157,7 +195,13 @@
     }}
   />
   {#if visible && !disabled}
-    <div class="picker" class:visible transition:fly={{ duration: 80, easing: cubicInOut, y: -5 }}>
+    <div
+      class="picker"
+      class:visible
+      transition:fly={{ duration: 80, easing: cubicInOut, y: -5 }}
```

Comment:
The transition here is broken when the positioning isn't default

Reply:
To resolve this I had to change it from reactive to beforeUpdate, If you are ok with if being a fade in or fade out instead of fly if could be moved back to reactive. the y position of the fly needs to be dynamically set or it looks wired if its not in the default location, and the transition gets run before reactive components.

Reply:
Can't it be reactive and set a `showAbove` variable that determines if the `y` is positive or negative?

Reply:
I see, to know if it should be shown above or below, it needs to check the picker element, but it only checks that element after the transition starts

Reply:
Figured it out using a custom transition that sets the position before running `fly()`